src/machineid.ts

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+import {commandSync} from 'execa';
+import {createHmac} from 'crypto';
+import * as os from 'os';
+
+function getCommand(): string {
+  switch (os.platform()) {
+    case 'darwin':
+        return 'ioreg -rd1 -c IOPlatformExpertDevice';
+    case 'win32':
+        return `${getWin32RegBinPath()}\\REG.exe ` +
+        'QUERY HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Cryptography ' +
+        '/v MachineGuid';
+    case 'linux':
+        return '( cat /var/lib/dbus/machine-id /etc/machine-id 2> /dev/null || hostname ) | head -n 1 || :';
+    case 'freebsd':
+        return 'kenv -q smbios.system.uuid || sysctl -n kern.hostuuid';
+    default:
+        throw new Error(`Unsupported platform: ${process.platform}`);
+  }
+}
+
+function getWin32RegBinPath(): string {
+  if(process.platform !== 'win32') {
+    return '';
+  }
+  
+  if( process.arch === 'ia32' && process.env.hasOwnProperty('PROCESSOR_ARCHITEW6432') ) {
+    return '%windir%\\sysnative\\cmd.exe /c %windir%\\System32';
+  }
+  
+  return '%windir%\\System32';
+}
+
+export function protect(): string {
+  let result = commandSync(getCommand(), {encoding: 'utf8'});
+  if (result.failed) {
+    return "na";
+  }
+
+  let id = expose(result.stdout);
```

Comment:
should also be const
```suggestion
  const id = expose(result.stdout);
```